sway/desktop/xdg_shell.c

```diff
@@ -217,7 +217,7 @@ static bool is_transient_for(struct sway_view *child,
 		return false;
 	}
 	struct wlr_xdg_surface *surface = child->wlr_xdg_surface;
-	while (surface) {
+	while (surface && surface->role != WLR_XDG_SURFACE_ROLE_NONE) {
```

Comment:
I think we could change these checks to be `surface->role == WLR_XDG_SURFACE_ROLE_TOPLEVEL`.